pkg/endpoint/metrics.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2016-2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+import (
+	"time"
+
+	"github.com/cilium/cilium/pkg/metrics"
+)
+
+const (
+	labelEndpointRegenerationBPFCompilation = "bpf-compilation"
+	labelEndpointProxyConfiguration         = "proxy-configuration"
+	labelEndpointRegenerationTotal          = "total"
+)
+
+// regenerationMetric is a helper function to update metrics.EndpointRegenerationTime
+func regenerationMetric(scope string, duration time.Duration) {
```

Comment:
Did you have other plans for this function? I would argue that leaving what this function does in ) `metricsEndpointRegeneration*` functions is actually more clear. In particular, if we have different labels for each then using this function becomes more complicated